Tasks/NuGetInstaller/NuGetInstaller.ps1

```diff
@@ -62,15 +65,40 @@ if(!$nuGetPath)
     $nuGetPath = Get-ToolPath -Name 'NuGet.exe';
 }
 
+if (-not $nugetPath)
+{
+    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+}
+
 if($nuGetRestoreArgs)
 {
+    if($nuGetRestoreArgs.ToLowerInvariant().Contains("-configfile"))
+    {
+        Write-Warning "ConfigFile was passed as a command line parameter, which may be ignored in certain parts of this task. Please specify the config file in the build definition instead."
+    }
+
     $args = ($args + " " + $nuGetRestoreArgs);
 }
 
-
-if (-not $nugetPath)
+if($nugetConfigPath -and ($nugetConfigPath -ne $env:Build_SourcesDirectory))
 {
-    throw (Get-LocalizedString -Key "Unable to locate {0}" -ArgumentList 'nuget.exe')
+    $args = "$args -configfile `"$tempNuGetConfigPath`""
+
+    $endpoint = Get-ServiceEndpoint -Context $distributedTaskContext -Name SystemVssConnection
+    if($endpoint.Authorization.Scheme -eq 'OAuth')
+    {
+        Write-Verbose "Getting credentials for $($endpoint)"
+        $accessToken = $endpoint.Authorization.Parameters['AccessToken']
+    }
+    else
+    {
+        Write-Warning "Could not determine credentials to use for NuGet"
```

Comment:
Let's touch base on this offline. I'd just like to understand more about this scenario where someone would use the nugetconfig param but not be using the oauth scheme. What's the auth scheme when building against an external git repo (e.g. github)?